syntax="proto3";
package Token.Service;

import "messages.proto";
import "blockchain.proto";

service BlockChainService{
    rpc GetHead (EmptyRequest) returns (Token.Messages.BlockHeader) {}
    rpc GetBlock (GetBlockRequest) returns (Token.Messages.BlockHeader) {}
    rpc GetTransactions (GetTransactionsRequest) returns (Token.Messages.TransactionList) {}
    rpc GetUnclaimedTransactions (GetUnclaimedTransactionsRequest) returns (Token.Messages.UnclaimedTransactionList);
    rpc AppendBlock (Token.Messages.Block) returns (Token.Messages.BlockHeader){}
}